languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,40 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="ACCORD_TOUT_ADJECTIF" name="accord tout adjectif" default="temp_off">
+          <rule>
+              <antipattern>
+                  <token>tout</token>
+                  <token regexp="yes">publique?|public?</token>
+              </antipattern>
+              <pattern>
+                  <token postag="(P\+)?D.*" postag_regexp="yes"/>
+                  <token postag="N f s" postag_regexp="yes"/>
+                  <marker>
+                      <token>tout</token>
+                  </marker>
+                  <token postag="J f s" postag_regexp="yes"><exception regexp="yes">^[aâeéèêioôœuhy].*</exception></token>
```

Comment:
Here and below, I would add the entity "h aspiré", otherwise words like "haute" will be treated as if they started with a vowel sound.
Whenever you use a `f` and a `s`, also think about `e` and `sp` – what shall we do with adjectives that remain the same for feminine and masculine (utile) or in singular and plural (honteux)?